storage/engine/bench_test.go

```diff
@@ -363,6 +363,51 @@ func runMVCCBatchPut(emk engineMaker, valueSize, batchSize int, b *testing.B) {
 	b.StopTimer()
 }
 
+func runMVCCBatchTimeSeries(emk engineMaker, batchSize int, b *testing.B) {
+	// Precompute keys so we don't waste time formatting them at each iteration.
+	numKeys := batchSize
+	keys := make([]roachpb.Key, numKeys)
+	for i := 0; i < numKeys; i++ {
+		keys[i] = roachpb.Key(fmt.Sprintf("key-%d", i))
+	}
+
+	var value roachpb.Value
+	if err := value.SetProto(&roachpb.InternalTimeSeriesData{
+		StartTimestampNanos: 0,
+		SampleDurationNanos: 1000,
+		Samples: []roachpb.InternalTimeSeriesSample{
+			{Offset: 0, Count: 1, Sum: 5.0},
+		},
+	}); err != nil {
+		b.Fatal(err)
+	}
+
+	stopper := stop.NewStopper()
+	eng, stopper := emk(b, fmt.Sprintf("batch_merge_%d", batchSize))
+	defer stopper.Stop()
+
+	b.ResetTimer()
+
+	ts := hlc.Timestamp{}
```

Comment:
This test is a little strange in how it will constantly write everything to the same sample bucket. That's not really how we write them in real life, but since we're only writing and never reading, shouldn't make any difference, though it's worth a note here.